### ISSUE TITLE
easyPlace wont work on servers in survival?

### ISSUE THREAD

**Issue by ShadowMaster13** [maruohon/litematica#49]
Block shadow will appear briefly however no block will place.

Litematica states "Action prevented by the Easy Place mode" # #

**Comment by maruohon** [comment]
It does work, unless the server is a Spigot/Paper variant or similar, which have cheat prevention and basically disallow right clicking on air blocks, which is how the Easy Place mode works currently.

I need to add another mode to it which would click on adjacent blocks, but that mode will then have even more issues with any blocks that have rotations, than what the current system already has.

Basically for proper rotation support of all blocks, there has to be a server side mod that can rotate the blocks according to what Litematica requests. Currently that basically means Carpet mod on the server, but even those don't currently work in 1.13.2 or 1.14, before I do some changes, and also I think Carpet has to again fully implement the Accurate Placement protocol it had in 1.12.

**Comment by ShadowMaster13** [comment]
If a block were to be seperately placed in air alone, would easyplace simply not function regardless of the current mode or the adjacent mode since air will be the only block in that space?

**Comment by maruohon** [comment]
Yep, Spigot/Paper simply won't allow a floating or only diagonally connected block to be placed, because there is nothing to click against to place it. So in other words, if it can't be placed in vanilla without temporary support blocks, then it's not possible to place it on Spigot/Paper servers using a printer or the Easy Place mode. And even if there are adjacent blocks, if the target block has a rotation, and that rotation will be wrong when the block is placed clicking on the existing adjacent block, then it would be placed in the wrong  rotation. On vanilla (or Forge or Fabric) because it's possible to click on air blocks, it's possible to also click on any block side of that air block, meaning that blocks such as Logs or Hoppers, which are placed depending on the block face you click on, *can* be placed correctly using the Easy Place mode (or could - currently it doesn't try to click on a specific side, I'm pretty sure).

**Comment by Domen-1** [comment]
Is there a setting in the paper that would remove this ban on placing blocks in the air?
